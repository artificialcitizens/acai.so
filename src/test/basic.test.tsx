import React from 'react';
import renderer from 'react-test-renderer';
import Link from '../components/Link';
import { test, expect } from 'vitest';

function toJson(component: renderer.ReactTestRenderer) {
  const result = component.toJSON();
  expect(result).toBeDefined();
  expect(result).not.toBeInstanceOf(Array);
  return result as renderer.ReactTestRendererJSON;
}

test('Link changes the class when hovered', () => {
  const component = renderer.create(<Link page="http://antfu.me">Anthony Fu</Link>);
  let tree = toJson(component);
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.props.onMouseEnter();

  // re-rendering
  tree = toJson(component);
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.props.onMouseLeave();
  // re-rendering
  tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
