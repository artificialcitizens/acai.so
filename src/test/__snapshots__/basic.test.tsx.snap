// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Link changes the class when hovered 1`] = `
<a
  className="normal"
  href="http://antfu.me"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Anthony Fu
</a>
`;

exports[`Link changes the class when hovered 2`] = `
<a
  className="hovered"
  href="http://antfu.me"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Anthony Fu
</a>
`;

exports[`Link changes the class when hovered 3`] = `
<a
  className="normal"
  href="http://antfu.me"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Anthony Fu
</a>
`;
